spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTable.java

```diff
@@ -341,4 +343,42 @@ public void testDowngradeTableToFormatV1ThroughTablePropertyFails() {
         "Cannot downgrade v2 table to v1",
         () -> sql("ALTER TABLE %s SET TBLPROPERTIES ('format-version'='1')", tableName));
   }
+
+  @Test
+  public void testCatalogSpecificWarehouseLocation() throws Exception {
+    File warehouseLocation = temp.newFolder(catalogName);
+    Assert.assertTrue(warehouseLocation.delete());
+    String location = "file:" + warehouseLocation.getAbsolutePath();
+    spark.conf().set("spark.sql.catalog." + catalogName + ".warehouse", location);
+    String testNameSpace =
+        (catalogName.equals("spark_catalog") ? "" : catalogName + ".") + "default1";
+    String testTableName = testNameSpace + ".table";
+    TableIdentifier newTableIdentifier = TableIdentifier.of("default1", "table");
+
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", testNameSpace);
+    sql("CREATE TABLE %s " + "(id BIGINT NOT NULL, data STRING) " + "USING iceberg", testTableName);
+
+    Table table = Spark3Util.loadIcebergCatalog(spark, catalogName).loadTable(newTableIdentifier);
+    if ("spark_catalog".equals(catalogName)) {
+      Assert.assertTrue(table.location().startsWith(spark.sqlContext().conf().warehousePath()));
+    } else {
+      Assert.assertTrue(table.location().startsWith(location));
+    }
+    sql("DROP TABLE IF EXISTS %s", testTableName);
+    sql("DROP NAMESPACE IF EXISTS %s", testNameSpace);
+  }
+
+  @Test
+  public void testCatalogWithSparkSqlWarehouseDir() {
+    spark.conf().unset("spark.sql.catalog." + catalogName + ".warehouse");
```

Comment:
Still not sure this is the write thing to do, if the sessionstate already has a catalog manager with the catalog init'd it would be used and this would have no effect.

Reply:
Thanks! I am also troubled by this problem, but it is not supported to start two SparkContexts in the same JVM, SparkSession getOrCreate method will return the existing spark instance. 
So this is why I rewrite the code and change this The reason why the test code is placed in a newly created `TestSparkWarehouseLocation`, because if it is placed in `TestCreateTable` to reset in different catalog warehouse config, once it fails, it will cause unit tests to affect each other, and pre-control Catalog state changes and recovery will complicate the overall `TestCreateTable` code, and these changes are unnecessary for other tests. The latest changes have been committed, what do you think?

Reply:
Looks like you figured it out. Like I was mentioning what you want to do in this case is make a brand new session (not a new context). You can have multiple sessions per spark context, each with different runtime configuration. So when we want to do something within a test to configure a session differently you can just call "newSession" at the beginning and use that new session in the test.